prow/cmd/checkconfig/main.go

```diff
@@ -1059,3 +1065,35 @@ func validateTideContextPolicy(cfg *config.Config) error {
 
 	return utilerrors.NewAggregate(errs)
 }
+
+func validateJobCluster(job config.JobBase) error {
+	if job.Cluster != "" && !(job.Agent == "" || job.Agent == string(v1.KubernetesAgent)) {
```

Comment:
That would be a lot easier to read as `(job.Agent != "" || job.Agent != string(v1.KubernetesAgent)`. Also, shouldn't this be opt in for the agents that do not support it? There is e.G. knative, no idea if it supports setting this field or ppl might write their own agent.

Reply:
TIL `agent: knative`.

I have not found any of it in test-infra jobs and (red hat) release jobs through.
If `knative` does not support `cluster`, we can added it into the set later.